cmd/thv-operator/controllers/virtualmcpserver_controller.go

```diff
@@ -0,0 +1,1071 @@
+	for _, pod := range podList.Items {
+		switch pod.Status.Phase {
+		case corev1.PodRunning:
+			running++
+		case corev1.PodPending:
+			pending++
+		case corev1.PodFailed:
+			failed++
+		case corev1.PodSucceeded:
+			running++
+		case corev1.PodUnknown:
+			pending++
+		}
+	}
+
+	// Update the status based on pod health
+	if running > 0 {
+		vmcp.Status.Phase = mcpv1alpha1.VirtualMCPServerPhaseReady
+		vmcp.Status.Message = "Virtual MCP server is running"
+		meta.SetStatusCondition(&vmcp.Status.Conditions, metav1.Condition{
+			Type:    mcpv1alpha1.ConditionTypeVirtualMCPServerReady,
+			Status:  metav1.ConditionTrue,
+			Reason:  "DeploymentReady",
+			Message: "Deployment is ready",
+		})
+	} else if pending > 0 {
+		vmcp.Status.Phase = mcpv1alpha1.VirtualMCPServerPhasePending
+		vmcp.Status.Message = "Virtual MCP server is starting"
+		meta.SetStatusCondition(&vmcp.Status.Conditions, metav1.Condition{
+			Type:    mcpv1alpha1.ConditionTypeVirtualMCPServerReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  "DeploymentNotReady",
+			Message: "Deployment is not yet ready",
+		})
+	} else if failed > 0 {
+		vmcp.Status.Phase = mcpv1alpha1.VirtualMCPServerPhaseFailed
+		vmcp.Status.Message = "Virtual MCP server failed to start"
+		meta.SetStatusCondition(&vmcp.Status.Conditions, metav1.Condition{
+			Type:    mcpv1alpha1.ConditionTypeVirtualMCPServerReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  "DeploymentFailed",
+			Message: "Deployment failed",
+		})
+	} else {
+		vmcp.Status.Phase = mcpv1alpha1.VirtualMCPServerPhasePending
+		vmcp.Status.Message = "No pods found for Virtual MCP server"
+		meta.SetStatusCondition(&vmcp.Status.Conditions, metav1.Condition{
+			Type:    mcpv1alpha1.ConditionTypeVirtualMCPServerReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  "DeploymentNotReady",
+			Message: "No pods found",
+		})
+	}
+
+	// Update ObservedGeneration to reflect that we've processed this generation
+	vmcp.Status.ObservedGeneration = vmcp.Generation
+
+	// Update status with conflict handling
+	// This is the final status update after all reconciliation steps complete
+	if err := r.Status().Update(ctx, vmcp); err != nil {
+		// Handle conflicts by returning error to trigger requeue
+		// Optimistic concurrency control: if another process updated the resource,
+		// we'll requeue and reconcile with the latest state
+		if errors.IsConflict(err) {
+			return err // Controller will requeue automatically
+		}
+		return fmt.Errorf("failed to update VirtualMCPServer status: %w", err)
+	}
+	return nil
+}
+
+// getHealthCheckInterval returns the health check interval for periodic reconciliation.
+// It parses the duration from the VirtualMCPServer spec if configured, otherwise returns the default.
+// Invalid duration strings fall back to the default. Intervals are clamped to min/max boundaries.
+func (*VirtualMCPServerReconciler) getHealthCheckInterval(vmcp *mcpv1alpha1.VirtualMCPServer) time.Duration {
```

Comment:
The healthcheck interval is something managed by the vmcp itself. I'm a little concerned about having the operator have constant reconciles based on the spec given by an operator. this could easily overload the controller. Can we remove this for now? I think we should instead have the vmcp have tighter integration with kubernetes and issue events if needed.